### PR TITLE
Renovate bot config

### PR DESCRIPTION
**Pull request by mayakoneval**
set up renovate bot using Trevor's template example

https://github.com/apollographql/typescript-repo-template#renovate

### CODE REVIEW

**Review by cheapsteak — APPROVED** [review]
👀

File: renovate.json5 [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      "groupSlug": "all-minor-patch",
+    },
+    // We set this to the lowest supported Node.js version to ensure we don't
+    // use newer Node.js APIs unknowingly during development which are going to
+    // fail in CI anyway when they're run against the full range of Node.js
+    // versions we support.
+    {
+      "matchPackageNames": ["@types/node"],
+      "allowedVersions": "12.x"
+    },
```
Comment (cheapsteak) [inline renovate.json5:27]: 💭 I don't think this applies to us since our code runs on the client and is transpiled

File: renovate.json5 [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  ],
```
Comment (mayakoneval) [inline renovate.json5:28]:
```suggestion

```

File: renovate.json5 [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  "packageRules": [
+    // Bunch up all non-major dependencies into a single PR.  In the common case
+    // where the upgrades apply cleanly, this causes less noise and is resolved faster
+    // than starting a bunch of upgrades in parallel for what may turn out to be
+    // a suite of related packages all released at once.
+    {
+      "groupName": "all non-major dependencies",
+      "matchUpdateTypes": ["patch", "minor"],
+      "groupSlug": "all-minor-patch",
+    },
```
Comment (mayakoneval) [inline renovate.json5:19]:
```suggestion
    },
  ]
```

File: renovate.json5 [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  ]
```
Comment (mayakoneval) [inline renovate.json5:20]:
```suggestion
  ],
```

**Pull request merged**